pkg/pod/status.go

```diff
@@ -135,51 +158,70 @@ func MakeTaskRunStatus(logger *zap.SugaredLogger, tr v1beta1.TaskRun, pod *corev
 			})
 		}
 	}
-
-	// Complete if we did not find a step that is not complete, or the pod is in a definitely complete phase
-	complete := areStepsComplete(pod) || pod.Status.Phase == corev1.PodSucceeded || pod.Status.Phase == corev1.PodFailed
-
-	if complete {
-		updateCompletedTaskRun(trs, pod)
-	} else {
-		updateIncompleteTaskRun(trs, pod)
-	}
+	trs.TaskRunResults = removeDuplicateResults(trs.TaskRunResults)
 
 	// Sort step states according to the order specified in the TaskRun spec's steps.
 	trs.Steps = sortTaskRunStepOrder(trs.Steps, taskSpec.Steps)
 
-	return *trs
+	return *trs, err
```

Comment:
Hm, it's ambiguous to me what the final returned error will be - it seems like it will be whatever the last call to `extractStartedAtTimeFromResults(results)` or `termination.ParseMessage(s.State.Terminated.Message)` returned.  In some cases this could be `nil` (since the last Step could Parse just fine) even if there were errors in earlier Steps?

It might be worth using the multierror package to aggregate together the complete list of errors?  See [this line from the taskrun reconciler](https://github.com/tektoncd/pipeline/blob/master/pkg/reconciler/taskrun/taskrun.go#L117) that shows usage of multierror.

Reply:
Good catch, I implemented multierrors now!